lib/explorer/data_frame.ex

```diff
@@ -1010,6 +1010,22 @@ defmodule Explorer.DataFrame do
         )
       )
 
+  @doc false
+  def filter_with(df, fun) when is_function(fun) do
+    ldf =
+      df
+      |> Explorer.Backend.LazyFrame.new()
+      |> Explorer.Backend.DataFrame.new(
+        df.names,
+        Enum.map(df.names, fn name -> df.dtypes[name] end)
```

Comment:
Hrm... why not pass `df.dtypes` directly?

Reply:
Yeah, I think we can change that :+1: